docs: Chapter 00：Connecting to the Interwebs (#231)
OpenINF/.github: doc/collections/_pages/content/setup--windows/chapter-0/Readme.md

---

> title: Connecting to the Interwebs (Wireless networking)

Author: ```suggestion
title: Connecting to the Interwebs (Wireless Networking)
```

---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm/
-->

> Before we can get started with development on our Windows Server machine, there are a few hurdles to overcome
> to have an actually-useful development environment.

Author: ```suggestion
Before we can begin using our Windows Server machine as an effective software
development environment, there are a few hurdles to overcome. This chapter deals
with enabling and configuring connectivity to the Internet.
```

---
title: Connecting to the Interwebs (Wireless Networking)
---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm/
-->

Before we can begin using our Windows Server machine as an effective software
development environment, there are a few hurdles to overcome. This chapter deals
with enabling and configuring connectivity to the Internet.

> ```

Author: ```suggestion

<!-- FIXME(DerekNonGeneric): Determine where from & document the below registry modification. -->

```text
```

reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\RadioManagement\SystemRadioState /ve /t REG_DWORD /d 0 /f
```

## Access to WiFi

> By default, access to WiFi is disabled in Windows Server 2022. This is because Airplane mode is enabled.
> It is also in an unconfigurable state (this preference cannot be changed by default). To be able to modify this
> preference, the following command must be run in an elevated Command Line Shell prompt.

Author: ```suggestion
As a safe default, on fresh installations of Windows Server, WiFi connectivity
is inaccessible because Airplane mode is enabled.

Airplane mode is also unconfigurable (the default preference of "enabled" cannot
be changed). The following command must be run in an elevated Command Line Shell
prompt to modify this preference.
```

Author: I think there may actually be two separate safe defaults that need to be overridden here and a step may be missing

Author: https://www.windowscentral.com/how-enable-or-disable-airplane-mode-windows-11#command-line-method-6

Author: > you may not be able to turn it on or off, or the feature may appear grayed out.

Author: https://www.windowscentral.com/how-enable-or-disable-airplane-mode-windows-11#command-line-method-5

Should we document creating a restore point too? Seems smart

---
title: Connecting to the Interwebs (Wireless Networking)

> ---

Author: ```suggestion
category: Documentation
content_type: Install & Setup
---
```

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github

> Spec: https://github.github.com/gfm/

Author: ```suggestion
Spec: https://github.github.com/gfm
```

---
title: Connecting to the Interwebs (Wireless Networking)
category: Documentation
content_type: Install & Setup
---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm
-->

> 

Author: ```suggestion

<!--
Not covered: Preparing device for (and installing) host OS --

-- Windows Server 2016/2022 Standard with Desktop Experience.

Windows Server 2016 Standard is for physical or minimally-virtualized environments.

See: https://www.microsoft.com/en-us/d/windows-server-2016-standard/dg7gmgf0ds12/0004
-->

```

---
title: Connecting to the Interwebs (Wireless Networking)
category: Documentation
content_type: Install & Setup
---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm
-->

<!-- Not covered: Preparing device for (and installing) host OS --
Windows Server 2016/2022 Standard with Desktop Experience.
Windows Server 2016 Standard is for physical or minimally-virtualized environments.

See: https://www.microsoft.com/en-us/d/windows-server-2016-standard/dg7gmgf0ds12/0004
-->

Before we can begin using our Windows Server machine as an effective software
development environment, there are a few hurdles to overcome. This chapter deals
with enabling and configuring connectivity to the Internet.

## Access to WiFi

As a safe default, on fresh installations of Windows Server, WiFi connectivity
is inaccessible because Airplane mode is enabled.

### Configuring Radio Management Service (RmSvc)[^1]

Airplane mode is also grayed-out/unconfigurable (the default preference of "enabled" cannot

> be changed). The following command must be run in an elevated Command Line Shell

Author: ```suggestion
be changed). The following commands must be run in an elevated Command Line Shell
```

prompt to modify this preference.

> 

Author: ```suggestion

<!-- FIXME(DerekNonGeneric):
Determine where from & document the below registry modification.
-->

```text
reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\RadioManagement\SystemRadioState /ve /t REG_DWORD /d 0 /f
```

```

---
title: Connecting to the Interwebs (Wireless Networking)
category: Documentation
content_type: Install & Setup
---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm
-->

<!-- Not covered: Preparing device for (and installing) host OS --
Windows Server 2016/2022 Standard with Desktop Experience.
Windows Server 2016 Standard is for physical or minimally-virtualized environments.

See: https://www.microsoft.com/en-us/d/windows-server-2016-standard/dg7gmgf0ds12/0004
-->

Before we can begin using our Windows Server machine as an effective software
development environment, there are a few hurdles to overcome. This chapter deals
with enabling and configuring connectivity to the Internet.

<!-- FIXME(DerekNonGeneric):
Determine where from & document the below registry modification.
-->

```text
reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\RadioManagement\SystemRadioState /ve /t REG_DWORD /d 0 /f
```

## Access to WiFi

As a safe default, on fresh installations of Windows Server, WiFi connectivity
is inaccessible because Airplane mode is enabled.

> Airplane mode is also unconfigurable (the default preference of "enabled" cannot

Author: ```suggestion

### Disabling Radio Management Service

Airplane mode is also grayed-out/unconfigurable (the default preference of "enabled" cannot
```

with enabling and configuring connectivity to the Internet.

<!-- FIXME(DerekNonGeneric):
Determine where from & document the below registry modification.
-->

```text
reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\RadioManagement\SystemRadioState /ve /t REG_DWORD /d 0 /f
```

## Access to WiFi

As a safe default, on fresh installations of Windows Server, WiFi connectivity
is inaccessible because Airplane mode is enabled.

### Disabling Radio Management Service (RmSvc)[^1]

Airplane mode is also grayed-out/unconfigurable (the default preference of "enabled" cannot
be changed). The following command must be run in an elevated Command Line Shell
prompt to modify this preference.

```cmd
SC CONFIG RmSvc START= AUTO
```

<!--
https://www.intel.com/content/www/us/en/download/19351/windows-10-and-windows-11-wi-fi-drivers-for-intel-wireless-adapters.html
https://support.lenovo.com/us/en/downloads/ds503062-fibocom-l850-gl-wireless-wan-driver-for-windows-10-version-1709-or-later-thinkpad
-->

```ps
Install-WindowsFeature -Name Wireless-Networking
```

reboot

```ps
shutdown –f –r –t 0
```

start service

```ps

> Start-Service WlanSvc –PassThru

Author: ```suggestion
Start-Service WlanSvc –PassThru
```

[^1]: https://github.com/MicrosoftDocs/windowsserverdocs/blob/main/WindowsServerDocs/security/windows-services/security-guidelines-for-disabling-system-services-in-windows-server.md#radio-management-service
```

---
title: Connecting to the Interwebs (Wireless Networking)
category: Documentation
content_type: Install & Setup
---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm
-->

<!-- Not covered: Preparing device for (and installing) host OS --
Windows Server 2016/2022 Standard with Desktop Experience.
Windows Server 2016 Standard is for physical or minimally-virtualized environments.

See: https://www.microsoft.com/en-us/d/windows-server-2016-standard/dg7gmgf0ds12/0004
-->

Before we can begin using our Windows Server machine as an effective software
development environment, there are a few hurdles to overcome. This chapter deals
with enabling and configuring connectivity to the Internet.

<!-- FIXME(DerekNonGeneric):
Determine where from & document the below registry modification.
-->

```text
reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\RadioManagement\SystemRadioState /ve /t REG_DWORD /d 0 /f
```

## Access to WiFi

As a safe default, on fresh installations of Windows Server, WiFi connectivity
is inaccessible because Airplane mode is enabled.

> ### Disabling Radio Management Service

Author: ```suggestion
### Disabling Radio Management Service (RmSvc)[^1]
```

Airplane mode is also grayed-out/unconfigurable (the default preference of "enabled" cannot
be changed). The following command must be run in an elevated Command Line Shell
prompt to modify this preference.

```cmd
SC CONFIG RmSvc START= AUTO
```

<!--
https://www.intel.com/content/www/us/en/download/19351/windows-10-and-windows-11-wi-fi-drivers-for-intel-wireless-adapters.html
https://support.lenovo.com/us/en/downloads/ds503062-fibocom-l850-gl-wireless-wan-driver-for-windows-10-version-1709-or-later-thinkpad
-->

```ps
Install-WindowsFeature -Name Wireless-Networking
```

reboot

```ps
shutdown –f –r –t 0
```

start service

```ps
Start-Service WlanSvc –PassThru

> ```

Author: ```suggestion
```

[^1]: https://github.com/MicrosoftDocs/windowsserverdocs/blob/main/WindowsServerDocs/security/windows-services/security-guidelines-for-disabling-system-services-in-windows-server.md#radio-management-service

```

---
title: Connecting to the Interwebs (Wireless Networking)
category: Documentation
content_type: Install & Setup
---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm
-->

<!-- Not covered: Preparing device for (and installing) host OS --

> Windows Server 2016/2022 Standard with Desktop Experience.

Author: ```suggestion

Windows Server 2016/2022 Standard with Desktop Experience.
```

---
title: Connecting to the Interwebs (Wireless Networking)
category: Documentation
content_type: Install & Setup
---

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm
-->

<!-- Not covered: Preparing device for (and installing) host OS --

Windows Server 2016/2022 Standard with Desktop Experience.
Windows Server 2016 Standard is for physical or minimally-virtualized environments.

See: https://www.microsoft.com/en-us/d/windows-server-2016-standard/dg7gmgf0ds12/0004

> -->

Author: ```suggestion

-->
```

Before we can begin using our Windows Server machine as an effective software
development environment, there are a few hurdles to overcome. This chapter deals
with enabling and configuring connectivity to the Internet.

> 
> 
> <!-- FIXME(DerekNonGeneric):
> Determine where from & document the below registry modification.
> -->
> 
> ```text
> reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\RadioManagement\SystemRadioState /ve /t REG_DWORD /d 0 /f
> ```

Author: ```suggestion
```

## Access to WiFi

As a safe default, on fresh installations of Windows Server, WiFi connectivity
is inaccessible because Airplane mode is enabled.

> ### Disabling Radio Management Service (RmSvc)[^1]

Author: ```suggestion
### Configuring Radio Management Service (RmSvc)[^1]
```

---
title: Connecting to the Interwebs (Wireless Networking)
category: Documentation
content_type: Install & Setup

> ---

Author: ```suggestion
published: false
---
```

<!-- Begin GitHub-Flavored Markdown (GFM)
See: https://docs.github.com/get-started/writing-on-github
Spec: https://github.github.com/gfm
-->

<!-- Not covered: Preparing device for (and installing) host OS --

Windows Server 2016/2022 Standard with Desktop Experience.
Windows Server 2016 Standard is for physical or minimally-virtualized environments.

See: https://www.microsoft.com/en-us/d/windows-server-2016-standard/dg7gmgf0ds12/0004

-->

Before we can begin using our Windows Server machine as an effective software
development environment, there are a few hurdles to overcome. This chapter deals
with enabling and configuring connectivity to the Internet.

> ## Access to WiFi

Author: ```suggestion
## Access to Wi-Fi
```

> As a safe default, on fresh installations of Windows Server, WiFi connectivity

Author: ```suggestion
As a safe default &mdash; on fresh installations of Windows Server &mdash; Wi-Fi connectivity
```
